tools/capability-checker/src/types/capability.ts

```diff
@@ -0,0 +1,34 @@
+import type {
+  LanguageModelV1,
+  ImageModelV1,
+  EmbeddingModelV1,
+} from '@ai-sdk/provider';
+
+export type CapabilityModelType = {
+  // Language model capabilities
+  textCompletion: 'language';
+  imageInput: 'language';
+  objectGeneration: 'language';
+  toolCalls: 'language';
+  pdfInput: 'language';
+  searchGrounding: 'language';
+  languageModelErrorHandling: 'language';
+  audioInput: 'language';
+
+  // Image model capabilities
+  imageGeneration: 'image';
+  imageModelErrorHandling: 'image';
+
+  // Embedding model capabilities
+  embedding: 'embedding';
+};
```

Comment:
I wonder if that's the best path or if we should split by model type.

Reply:
i.e. `type LanguageModelCapabilities = "textCompletion" | "imageInput" | ...` (or have as array and derive type)